pkg/scheduler/plugins/elasticquota/core/group_quota_manager.go

```diff
@@ -0,0 +1,616 @@
+		return
+	}
+	if treeWrapperPtr.NeedUpdateOneGroupRequest(quotaInfo) {
+		treeWrapperPtr.UpdateOneGroupRequest(quotaInfo)
+	}
+
+	for i := 1; i < len(curToAllParInfos); i++ {
+		parentQuotaInfo := curToAllParInfos[i]
+		directParTreeWrapperPtr := gqm.getTreeWrapperByNameNoLock(parentQuotaInfo.ParentName)
+		if directParTreeWrapperPtr == nil {
+			klog.Errorf("treeWrapper not exist!  parentName:%v", quotaInfo.Name, quotaInfo.ParentName)
+			return
+		}
+
+		oldSubLimitReqTmp := quotav1.Subtract(v1.ResourceList{}, oldSubLimitReq)
+		oldSubLimitReq = parentQuotaInfo.getLimitRequestNoLock()
+		parentQuotaInfo.addRequestNonNegativeNoLock(oldSubLimitReqTmp)
+		parentQuotaInfo.addRequestNonNegativeNoLock(newSubLimitReq)
+		newSubLimitReq = parentQuotaInfo.getLimitRequestNoLock()
+
+		if directParTreeWrapperPtr.NeedUpdateOneGroupRequest(parentQuotaInfo) {
+			directParTreeWrapperPtr.UpdateOneGroupRequest(parentQuotaInfo)
+		}
+	}
+}
+
+func (gqm *GroupQuotaManager) UpdateGroupDeltaUsed(quotaName string, delta v1.ResourceList) {
+	gqm.Lock.RLock()
+	defer gqm.Lock.RUnlock()
+
+	gqm.updateGroupDeltaUsedNoLock(quotaName, delta)
+	// if systemQuotaGroup or DefaultQuotaGroup's used change, update cluster total resource.
+	if quotaName == extension.SystemQuotaName || quotaName == extension.DefaultQuotaName {
+		gqm.updateClusterTotalResourceNoLock(v1.ResourceList{})
+	}
+}
+
+// updateGroupDeltaUsedNoLock updates the usedQuota of a node, it also updates all parent nodes
+// no need to lock gqm.lock
+func (gqm *GroupQuotaManager) updateGroupDeltaUsedNoLock(quotaName string, delta v1.ResourceList) {
+	curToAllParInfos := gqm.getCurToAllParentGroupQuotaInfoNoLock(quotaName)
+	allQuotaInfoLen := len(curToAllParInfos)
+	if allQuotaInfoLen <= 0 {
+		return
+	}
+
+	defer gqm.scopedLockForQuotaInfo(curToAllParInfos)()
+	for i := 0; i < allQuotaInfoLen; i++ {
+		quotaInfo := curToAllParInfos[i]
+		quotaInfo.addUsedNonNegativeNoLock(delta)
+	}
+}
+
+func (gqm *GroupQuotaManager) RefreshRuntime(quotaName string) v1.ResourceList {
+	gqm.Lock.RLock()
+	defer gqm.Lock.RUnlock()
+
+	return gqm.refreshRuntimeNoLock(quotaName)
+}
+
+func (gqm *GroupQuotaManager) refreshRuntimeNoLock(quotaName string) v1.ResourceList {
+	quotaInfo := gqm.getGroupQuotaInfoByNameNoLock(quotaName)
+	if quotaInfo == nil {
+		return nil
+	}
+
+	if quotaName == extension.SystemQuotaName || quotaName == extension.DefaultQuotaName {
+		return quotaInfo.GetMax()
+	}
+
+	return gqm.refreshAndGetRuntimeQuotaNoLock(quotaInfo)
+}
+
+// refreshAndGetRuntimeQuotaNoLock no need to lock gqm.lock
+func (gqm *GroupQuotaManager) refreshAndGetRuntimeQuotaNoLock(quotaInfo *QuotaInfo) v1.ResourceList {
+	curToAllParInfos := gqm.getCurToAllParentGroupQuotaInfoNoLock(quotaInfo.Name)
+	quotaInfoLen := len(curToAllParInfos)
+	if quotaInfoLen == 0 {
+		return nil
+	}
+
+	defer gqm.scopedLockForQuotaInfo(curToAllParInfos)()
+
+	if quotaInfoLen == 1 {
```

Comment:
this is "if else" can be merged. 

 for i := quotaInfoLen - 1; i >= 0; i-- {
     1.update current i's runtime
     2. if i -1 >= 0, continue update sub's total
 }